py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,708 @@
+            'k': {
+                'cur': round(kelvin['temp']),
+                'max': round(kelvin['temp_max']),
+                'min': round(kelvin['temp_min'])}}
+
+        # Get the choice and add more
+        choice = options[self.temp_unit]
+        choice['icon'] = self.icons['temp']
+        format_str = self.format_temp
+
+        # Optionally add the color
+        color = None
+        if self.temp_color:
+            color = self.py3.threshold_get_color(choice['cur'])
+            color_str = '\?color=%s' % color
+            format_str = color_str + ' ' + format_str
+
+        # Format the temperature
+        return self.py3.safe_format(format_str, choice)
+
+    def _format_sunrise(self, wthr):
+        # Get the time for sunrise (default is the start of time)
+        dt = datetime.datetime.utcfromtimestamp(
+            self._jpath(wthr, OWM_SUNRISE, 0))
+
+        # Format the sunrise
+        replaced = dt.strftime(self.format_sunrise)
+        return self.py3.safe_format(replaced, {
+            'icon': self.icons['sunrise'],
+        })
+
+    def _format_sunset(self, wthr):
+        # Get the time for sunset (default is the start of time)
+        dt = datetime.datetime.utcfromtimestamp(
+            self._jpath(wthr, OWM_SUNSET, 0))
+
+        # Format the sunset
+        replaced = dt.strftime(self.format_sunset)
+        return self.py3.safe_format(replaced, {
+            'icon': self.icons['sunset'],
+        })
+
+    def _format_dict(self, wthr):
+        return {
+            # Standard options
+            'icon': self._get_icon(wthr),
+            'clouds': self._format_clouds(wthr),
+            'rain': self._format_rain(wthr),
+            'snow': self._format_rain(wthr),
+            'wind': self._format_wind(wthr),
+            'humidity': self._format_humidity(wthr),
+            'pressure': self._format_pressure(wthr),
+            'temp': self._format_temp(wthr),
+            'sunrise': self._format_sunrise(wthr),
+            'sunset': self._format_sunset(wthr),
+
+            # Descriptions (defaults to empty)
+            'desc': self._jpath(wthr, OWM_DESC, '').lower(),
+            'desc_long': self._jpath(wthr, OWM_DESC_LONG, '')
+        }
+
+    def _format(self, wthr, fcsts):
+        # Format all sections
+        today = self._format_dict(wthr)
+
+        # Insert forecasts
+        forecasts = []
+        for day in fcsts:
+            future = self._format_dict(day)
+            forecasts.append(self.py3.safe_format(self.format_forecast, future))
+
+        # Give the final format
+        today['forecast'] = self.py3.composite_join(
+            self.forecast_text_separator, forecasts)
+        return self.py3.safe_format(self.format, today)
+
+    def weather(self):
+        # Get weather information
+        coords = self._get_coords()
+        text = ''
+        if coords is not None:
+            wthr = self._get_weather(coords)
+            fcsts = self._get_forecast(coords)
```

Comment:
Here, I think... We make double requests on every loop... only when we have the forecasts enabled.
Should we try and see if we can obtain both `wthr` and `fcsts` with one `_make_req()` instead?